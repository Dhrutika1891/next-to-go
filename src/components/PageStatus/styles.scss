.page-status {
    text-align: center;
}