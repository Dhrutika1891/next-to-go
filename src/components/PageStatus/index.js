import React from "react";
import { Spinner } from "react-bootstrap";

import "./styles.scss";

function PageStatus({ status }) {
  return (
    <div className="page-status" data-testid="page-status">
      {status === "loading" && (
        <div data-testid="status-loading">
          <Spinner animation="border" variant="primary" />
          <div>Loading...</div>
        </div>
      )}
      {status === "empty" && (
        <p data-testid="status-empty">
          There was no result for the page. Please refresh the page or contact
          support Team.
        </p>
      )}
      {status === "error" && (
        <p data-testid="status-error">
          There was an error loading the page. Please refresh the page or
          contact support Team.
        </p>
      )}
    </div>
  );
}

export default PageStatus;
