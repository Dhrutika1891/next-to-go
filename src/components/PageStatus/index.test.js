import { render } from "@testing-library/react";
import React from "react";

import PageStatus from ".";

describe("component page status", () => {
  it("should render page status", () => {
    expect.hasAssertions();

    const { getByTestId, baseElement } = render(
      <PageStatus status="loading" />
    );

    const pageStatus = getByTestId("page-status");
    expect(pageStatus).toBeInTheDocument();

    expect(baseElement).toMatchSnapshot();
  });

  it("should render page status loading", () => {
    expect.hasAssertions();

    const { getByTestId, queryByTestId, baseElement } = render(
      <PageStatus status="loading" />
    );

    const loadingStatus = getByTestId("status-loading");
    expect(loadingStatus).toBeInTheDocument();
    expect(loadingStatus).toHaveTextContent("Loading");

    const emptyStatus = queryByTestId("status-empty");
    expect(emptyStatus).not.toBeInTheDocument();

    const errorStatus = queryByTestId("status-error");
    expect(errorStatus).not.toBeInTheDocument();

    expect(baseElement).toMatchSnapshot();
  });

  it("should render page status empty", () => {
    expect.hasAssertions();

    const { getByTestId, queryByTestId, baseElement } = render(
      <PageStatus status="empty" />
    );

    const loadingStatus = queryByTestId("status-loading");
    expect(loadingStatus).not.toBeInTheDocument();

    const emptyStatus = getByTestId("status-empty");
    expect(emptyStatus).toBeInTheDocument();

    const errorStatus = queryByTestId("status-error");
    expect(errorStatus).not.toBeInTheDocument();

    expect(baseElement).toMatchSnapshot();
  });

  it("should render page status error", () => {
    expect.hasAssertions();

    const { getByTestId, queryByTestId, baseElement } = render(
      <PageStatus status="error" />
    );

    const loadingStatus = queryByTestId("status-loading");
    expect(loadingStatus).not.toBeInTheDocument();

    const emptyStatus = queryByTestId("status-empty");
    expect(emptyStatus).not.toBeInTheDocument();

    const errorStatus = getByTestId("status-error");
    expect(errorStatus).toBeInTheDocument();

    expect(baseElement).toMatchSnapshot();
  });
});
