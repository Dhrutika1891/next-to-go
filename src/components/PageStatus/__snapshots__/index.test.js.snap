// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component page status should render page status 1`] = `
<body>
  <div>
    <div
      class="page-status"
      data-testid="page-status"
    >
      <div
        data-testid="status-loading"
      >
        <div
          class="spinner-border text-primary"
        />
        <div>
          Loading...
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`component page status should render page status empty 1`] = `
<body>
  <div>
    <div
      class="page-status"
      data-testid="page-status"
    >
      <p
        data-testid="status-empty"
      >
        There was no result for the page. Please refresh the page or contact support Team.
      </p>
    </div>
  </div>
</body>
`;

exports[`component page status should render page status error 1`] = `
<body>
  <div>
    <div
      class="page-status"
      data-testid="page-status"
    >
      <p
        data-testid="status-error"
      >
        There was an error loading the page. Please refresh the page or contact support Team.
      </p>
    </div>
  </div>
</body>
`;

exports[`component page status should render page status loading 1`] = `
<body>
  <div>
    <div
      class="page-status"
      data-testid="page-status"
    >
      <div
        data-testid="status-loading"
      >
        <div
          class="spinner-border text-primary"
        />
        <div>
          Loading...
        </div>
      </div>
    </div>
  </div>
</body>
`;
